R/provider-aws.R

```diff
@@ -208,13 +228,32 @@ method(chat_request, ProviderAWSBedrock) <- function(
     toolConfig <- NULL
   }
 
+  # Get standardized params and merge with manual api_args
+  params <- chat_params(provider, provider@params)
+
+  # Merge params into inferenceConfig, giving precedence to manual api_args
+  inference_config <- modify_list(
```

Comment:
Would it be possible to ensure that this is `NULL` when there are no params? Then you can add in `body` and use `compact()` like we do in other places.

Reply:
I took a pass at this and ended up with an overall more compact version:

```r
  # Merge params into inferenceConfig, giving precedence to manual api_args
  params <- chat_params(provider, provider@params)

  extra_args <- provider@extra_args
  extra_args$inferenceConfig <- modify_list(
    params,
    extra_args$inferenceConfig %||% list()
  )

  # https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_Converse.html
  body <- compact(dots_list(
    messages = messages,
    system = system,
    toolConfig = toolConfig,
    !!!extra_args,
    .homonyms = "first"
  ))
```